spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java

```diff
@@ -168,19 +172,105 @@ public void setExtendedBindingProperties(RabbitExtendedBindingProperties extende
+					this.rabbitProperties.getUsername(), this.rabbitProperties.getPassword(),
+					this.rabbitProperties.getSsl().isEnabled(), this.rabbitProperties.getSsl().getKeyStore(),
+					this.rabbitProperties.getSsl().getTrustStore(),
+					this.rabbitProperties.getSsl().getKeyStorePassword(),
+					this.rabbitProperties.getSsl().getTrustStorePassword());
+		}
+	}
+
+	/**
+	 * @see org.springframework.boot.autoconfigure.amqp.RabbitAutoConfiguration.RabbitConnectionFactoryCreator
+	 */
+	private CachingConnectionFactory createProducerConnectionFactory(RabbitProperties config) throws Exception {
+		RabbitConnectionFactoryBean factory = new RabbitConnectionFactoryBean();
+		if (config.determineHost() != null) {
+			factory.setHost(config.determineHost());
+		}
+		factory.setPort(config.determinePort());
+		if (config.determineUsername() != null) {
+			factory.setUsername(config.determineUsername());
+		}
+		if (config.determinePassword() != null) {
+			factory.setPassword(config.determinePassword());
+		}
+		if (config.determineVirtualHost() != null) {
+			factory.setVirtualHost(config.determineVirtualHost());
+		}
+		if (config.getRequestedHeartbeat() != null) {
+			factory.setRequestedHeartbeat(config.getRequestedHeartbeat());
+		}
+		RabbitProperties.Ssl ssl = config.getSsl();
+		if (ssl.isEnabled()) {
+			factory.setUseSSL(true);
+			if (ssl.getAlgorithm() != null) {
+				factory.setSslAlgorithm(ssl.getAlgorithm());
+			}
+			factory.setKeyStore(ssl.getKeyStore());
+			factory.setKeyStorePassphrase(ssl.getKeyStorePassword());
+			factory.setTrustStore(ssl.getTrustStore());
+			factory.setTrustStorePassphrase(ssl.getTrustStorePassword());
+		}
+		if (config.getConnectionTimeout() != null) {
+			factory.setConnectionTimeout(config.getConnectionTimeout());
+		}
+		factory.afterPropertiesSet();
+		CachingConnectionFactory connectionFactory = new CachingConnectionFactory(factory.getObject());
+		connectionFactory.setAddresses(config.determineAddresses());
+		connectionFactory.setPublisherConfirms(config.isPublisherConfirms());
+		connectionFactory.setPublisherReturns(config.isPublisherReturns());
+		if (config.getCache().getChannel().getSize() != null) {
+			connectionFactory
+					.setChannelCacheSize(config.getCache().getChannel().getSize());
+		}
+		if (config.getCache().getConnection().getMode() != null) {
+			connectionFactory
+					.setCacheMode(config.getCache().getConnection().getMode());
+		}
+		if (config.getCache().getConnection().getSize() != null) {
+			connectionFactory.setConnectionCacheSize(
+					config.getCache().getConnection().getSize());
+		}
+		if (config.getCache().getChannel().getCheckoutTimeout() != null) {
+			connectionFactory.setChannelCheckoutTimeout(
+					config.getCache().getChannel().getCheckoutTimeout());
+		}
+		return connectionFactory;
+	}
+
+	@Override
+	public void destroy() throws Exception {
+		if (this.connectionFactory instanceof DisposableBean) {
+			((DisposableBean) this.connectionFactory).destroy();
```

Comment:
I don't think we should destroy() this one since it's a bean injected into the CTOR.

Reply:
First of all we may override it into the `LocalizedQueueConnectionFactory`, which may spawn its internal conection factories. OTOH the `destroy()` in the `CCF` looks idempotent